Sources/Fullscreen/AddressMapView/AddressMapView.swift

```diff
@@ -0,0 +1,207 @@
+    public weak var delegate: AddressMapViewDelegate?
+
+    // MARK: - Private properties
+
+    private var tileOverlay: MKTileOverlay?
+    private var circle: MKCircle?
+    private var annotation: MKAnnotation?
+    private var polygons = [MKPolygon]()
+
+    private lazy var mapView: MKMapView = {
+        let view = MKMapView(withAutoLayout: true)
+        view.isRotateEnabled = false
+        view.isPitchEnabled = false
+        view.delegate = self
+        return view
+    }()
+
+    private lazy var viewModeButton: UIButton = {
+        let button = UIButton.mapButton
+        button.setImage(UIImage(named: .viewMode).withRenderingMode(.alwaysTemplate), for: .normal)
+        button.addTarget(self, action: #selector(handleViewModeButtonTap), for: .touchUpInside)
+        return button
+    }()
+
+    private lazy var pinButton: UIButton = {
+        let button = UIButton.mapButton
+        button.setImage(UIImage(named: .pin).withRenderingMode(.alwaysTemplate), for: .normal)
+        button.addTarget(self, action: #selector(handlePinButtonTap), for: .touchUpInside)
+        return button
+    }()
+
+    // MARK: - Init
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        fatalError()
+    }
+
+    // MARK: - Public API
+
+    public func centerMap(location: CLLocationCoordinate2D, regionDistance: Double, animated: Bool) {
+        let meters: CLLocationDistance = regionDistance
+        let region = MKCoordinateRegion(center: location, latitudinalMeters: meters, longitudinalMeters: meters)
+        mapView.setRegion(region, animated: animated)
+    }
+
+    public func configureTileOverlay(_ tileOverlay: MKTileOverlay) {
+        if let oldOverlay = self.tileOverlay {
+            mapView.removeOverlay(oldOverlay)
+        }
+        mapView.addOverlay(tileOverlay, level: .aboveLabels)
+        self.tileOverlay = tileOverlay
+    }
+
+    public func configureRadiusArea(_ radius: Double, location: CLLocationCoordinate2D) {
+        removeCurrentAnnotationAndShapeOverlays()
+        let newCircle = MKCircle(center: location, radius: radius)
+        mapView.addOverlay(newCircle)
+        circle = newCircle
+    }
+
+    public func configureAnnotation(title: String, location: CLLocationCoordinate2D) {
+        removeCurrentAnnotationAndShapeOverlays()
+        let newAnnotation = AddressAnnotation(title: title, location: location)
+        mapView.addAnnotation(newAnnotation)
+        annotation = newAnnotation
+    }
+
+    public func makePolygonOverlayVisible() {
+        guard let firstPolygon = polygons.first else {
+            return
+        }
+
+        let mapRect = polygons.dropFirst().reduce(firstPolygon.boundingMapRect, { (mapRect, polygon) -> MKMapRect in
+            return polygon.boundingMapRect.union(mapRect)
+        })
+        let bottomInset = UIDevice.isIPad() ? 16 : 16 + .mediumLargeSpacing
```

Comment:
Should we start using `traitCollection.verticalSizeClass.regular` for these types of checks?

Reply:
Definitely 👍